src/binding/petsc4py/src/petsc4py/PETSc/DMDA.pyx

```diff
@@ -171,7 +328,21 @@ cdef class DMDA(DM):
                             NULL) )
         return toDims(dim, M, N, P)
 
-    def setProcSizes(self, proc_sizes):
+    def setProcSizes(self, proc_sizes: tuple[int, ...]) -> None:
+        """Set the number of processes in each dimension.
```

Comment:
It seems you use `dimensions` and `directions` almost interchangeably. I'm ok with both, but we have to be consistent and use only one.